spec/API_specification/array_api/array_object.py

```diff
@@ -224,19 +224,40 @@ def __array_namespace__(self: array, /, *, api_version: Optional[str] = None) ->
 
     def __bool__(self: array, /) -> bool:
         """
-        Converts a zero-dimensional boolean array to a Python ``bool`` object.
+        Converts a zero-dimensional array to a Python ``bool`` object.
+
+        .. note::
+           When casting a real-valued input array to ``bool``, a value of ``0`` must cast to ``False``, and a non-zero value must cast to ``True``.
 
         Parameters
         ----------
         self: array
-            zero-dimensional array instance. Must have a boolean data type.
+            zero-dimensional array instance.
 
         Returns
         -------
         out: bool
             a Python ``bool`` object representing the single element of the array.
         """
 
+    def __complex__(self: array, /) -> complex:
```

Comment:
Are complex numbers going to be required or optional in the specification? If they are optional this isn't needed for real arrays (`complex()` calls `__float__` if `__complex__` is not defined).

Reply:
Required right? I might of missed discussions on opt-in complex support.

Reply:
Required.